Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -101,34 +111,37 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
                         change:(NSDictionary<NSKeyValueChangeKey, id> *)change
                        context:(void *)context
 {
-    if ([keyPath isEqualToString:NSStringFromSelector(@selector(state))]) {
-        NSURLSessionTask *sessionTask = object;
-        if (sessionTask.state != NSURLSessionTaskStateRunning) {
-            SentrySpanId *spanId;
-            @synchronized(sessionTask) {
-                spanId = objc_getAssociatedObject(
-                    sessionTask, &SENTRY_NETWORK_REQUEST_TRACKER_SPAN_ID);
-                // We'll just go through once
-                objc_setAssociatedObject(sessionTask, &SENTRY_NETWORK_REQUEST_TRACKER_SPAN_ID, nil,
-                    OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-            }
-
-            if (spanId != nil) {
-                id<SentrySpan> span = [self.tracker getSpan:spanId];
-                NSInteger responseStatusCode = [self urlResponseStatusCode:sessionTask.response];
-
-                if (responseStatusCode != -1) {
-                    [span setDataValue:[NSNumber numberWithInteger:responseStatusCode]
-                                forKey:@"http.status_code"];
-                }
-
-                [self.tracker finishSpan:spanId withStatus:[self statusForSessionTask:sessionTask]];
-                [sessionTask removeObserver:self forKeyPath:NSStringFromSelector(@selector(state))];
-                [SentryLog logWithMessage:@"Finished HTTP span for sessionTask"
-                                 andLevel:kSentryLevelDebug];
-            }
-        }
+    if (![keyPath isEqualToString:NSStringFromSelector(@selector(state))]) {
```

Comment:
`m`: I changed the structure here a bit [to use guard clauses to reduce nesting](https://www.refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html). This reveals that we don't have tests for the return statements here. Can we please add some tests to get the coverage up to 100%.